src/minio.js

```diff
@@ -598,153 +595,12 @@ export class Client extends TypedClient {
-          var partsDone = []
-          var partNumber = 1
-          var uploadedSize = 0
-          async.whilst(
-            (cb) => {
-              cb(null, uploadedSize < size)
-            },
-            (cb) => {
-              var stream
-              var cbTriggered = false
-              var origCb = cb
-              cb = function () {
-                if (cbTriggered) {
-                  return
-                }
-                cbTriggered = true
-                if (stream) {
-                  stream.destroy()
-                }
-                return origCb.apply(this, arguments)
-              }
-              var part = parts[partNumber]
-              var hash = transformers.getHashSummer(this.enableSHA256)
-              var length = partSize
-              if (length > size - uploadedSize) {
-                length = size - uploadedSize
-              }
-              var start = uploadedSize
-              var end = uploadedSize + length - 1
-              var autoClose = true
-              var options = { autoClose, start, end }
-              // verify md5sum of each part
-              pipesetup(fs.createReadStream(filePath, options), hash)
-                .on('data', (data) => {
-                  var md5sumHex = Buffer.from(data.md5sum, 'base64').toString('hex')
-                  if (part && md5sumHex === part.etag) {
-                    // md5 matches, chunk already uploaded
-                    partsDone.push({ part: partNumber, etag: part.etag })
-                    partNumber++
-                    uploadedSize += length
-                    return cb()
-                  }
-                  // part is not uploaded yet, or md5 mismatch
-                  stream = fs.createReadStream(filePath, options)
-                  uploader(uploadId, partNumber, stream, length, data.sha256sum, data.md5sum, (e, objInfo) => {
-                    if (e) {
-                      return cb(e)
-                    }
-                    partsDone.push({ part: partNumber, etag: objInfo.etag })
-                    partNumber++
-                    uploadedSize += length
-                    return cb()
-                  })
-                })
-                .on('error', (e) => cb(e))
-            },
-            (e) => {
-              if (e) {
-                return cb(e)
-              }
-              cb(null, partsDone, uploadId)
-            },
-          )
-        },
-        // all parts uploaded, complete the multipart upload
-        (etags, uploadId, cb) => this.completeMultipartUpload(bucketName, objectName, uploadId, etags, cb),
-      ],
-      (err, ...rest) => {
-        if (err === true) {
-          return
-        }
-        callback(err, ...rest)
-      },
-    )
+    fs.lstat(filePath, (err, stat) => {
+      if (err) {
+        return callback(err)
+      }
+      return this.putObject(bucketName, objectName, fs.createReadStream(filePath), stat.size, metaData, callback)
+    })
```

Comment:
This PR brought changes from https://github.com/minio/minio-js/pull/1159

Reply:
Oh, sorry. Should wait that PR.